python_modules/dagster/dagster/_core/storage/fs_io_manager.py

```diff
@@ -199,6 +199,24 @@ def handle_output(self, context, obj):
 
     def load_input(self, context):
         """Unpickle the file and Load it to a data object."""
+        if context.has_asset_partitions:
+            return [
+                self._load_pickle(self._partition_path(context.asset_key, partition_key))
+                for partition_key in context.asset_partition_keys
+            ]
+        else:
+            return self._load_input(context)
+
+    def _partition_path(self, asset_key, partition_key):
```

Comment:
Not required, but if you feel up to adding type annotations to the argument types and return types, it would be helpful

Reply:
I've added type annotations, did you want me to take this opportunity to add more type annotations to the signature of `load_input` or is it sufficient to just type annotate the methods I added?

Reply:
Annotating the methods you added is great!